addons/web/static/src/js/framework/misc.js

```diff
@@ -204,6 +204,39 @@ function ReloadContext (parent, action) {
 }
 core.action_registry.add("reload_context", ReloadContext);
 
+
+/**
+ * Generic widget to create an iframe inside the backend
+ * that listens for clicks to interact with the backend.
+ *
+ * It should be extended by overwritting the methods:
+ *      init: function(parent) {
+ *          this._super(parent, <url_of_iframe>
+ *      },
+ *      iframe_clicked: function(e){
+ *          filter the clicks you want to use and apply
+ *          an action on it
+ *      }
+ */
+var IframeBackend = Widget.extend({
```

Comment:
i don't like the name. there is no reason to tie the name to the backend, especially since backend is a confusing term in odoo (everywhere else, backend is the server).  You can simply name it IFrameWidget